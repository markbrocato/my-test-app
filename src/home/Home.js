import React, { Component } from 'react'
import { inject, observer } from 'mobx-react'
import Container from 'react-storefront/Container'
import Typography from '@material-ui/core/Typography'
import Row from 'react-storefront/Row'
import CmsSlot from 'react-storefront/CmsSlot'
import withAmp from 'react-storefront/amp/withAmp'

@withAmp
@inject('app')
@observer
export default class Home extends Component { 

  render() {
    const { app } = this.props

    return (
      <Container> 
        <Row>
          <Typography variant="h4">React Storefront</Typography>
        </Row>
        <Row>
          <Typography variant="subtitle1">
            <CmsSlot>{ app.welcomeMessage }</CmsSlot>
          </Typography>
        </Row>
      </Container> 
    )
  }

}
