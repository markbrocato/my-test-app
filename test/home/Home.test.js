import React from 'react'
import { mount } from 'enzyme'
import Home from '../../src/home/Home'
import TestProvider from '../TestProvider'

describe('Home', () => {
  it('should render', () => {
    expect(mount(
      <TestProvider>
        <Home/>
      </TestProvider>
    )).toMatchSnapshot()
  })
})