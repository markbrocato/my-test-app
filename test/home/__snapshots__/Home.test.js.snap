// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<TestProvider>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <inject-c-with-app-history>
        <c app={{...}} history={{...}}>
          <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
            <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} link={{...}}>
              <NullComponent defer={true} encodeSpecialCharacters={true} link={{...}} />
            </SideEffect(NullComponent)>
          </HelmetWrapper>
          <inject-Home-with-app app={{...}} history={{...}}>
            <Home app={{...}} history={{...}}>
              <WithStyles(b)>
                <b classes={{...}} inset={true}>
                  <div className="RSFContainer-root-1 RSFContainer-inset-2">
                    <WithStyles(b)>
                      <b classes={{...}}>
                        <div className="RSFRow-root-3">
                          <WithStyles(Typography) variant="h4">
                            <Typography variant="h4" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                              <h4 className="MuiTypography-root-4 MuiTypography-h4-19">
                                React Storefront
                              </h4>
                            </Typography>
                          </WithStyles(Typography)>
                        </div>
                      </b>
                    </WithStyles(b)>
                    <WithStyles(b)>
                      <b classes={{...}}>
                        <div className="RSFRow-root-3">
                          <WithStyles(Typography) variant="subtitle1">
                            <Typography variant="subtitle1" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                              <h6 className="MuiTypography-root-4 MuiTypography-subtitle1-22">
                                <WithStyles(b)>
                                  <b classes={{...}} />
                                </WithStyles(b)>
                              </h6>
                            </Typography>
                          </WithStyles(Typography)>
                        </div>
                      </b>
                    </WithStyles(b)>
                  </div>
                </b>
              </WithStyles(b)>
            </Home>
          </inject-Home-with-app>
        </c>
      </inject-c-with-app-history>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;
